lineage/BaumWelch.py

```diff
@@ -145,6 +148,33 @@ def do_M_E_step(tHMMobj, gammas):
         tHMMobj.estimate.E[state_j].estimator(all_cells, all_gammas[:, state_j])
 
 
+def do_M_E_step_1state(all_tHMMobj, gammas_1st, state_j):
+    """ perform the M_E step when all the concentrations are given at once for one state.
+    all_cells is a list (for different concentrations) containing lists of tuples of observations. 
+    gammas_1st is a list of lists, each cell's gamma value for one state. """
+    all_cells = []
+    for tHMMobj in all_tHMMobj:
+        all_cells.append([cell.obs for lineage in tHMMobj.X for cell in lineage.output_lineage])
+        # TODO: the line where we pass the cells and gammas to the atonce_estimator
+        for i in range(len(tHMMobj.estimate.E[state_j].params)):
+            tHMMobj.estimate.E[state_j].params[i] = 0.5
+
+def do_M_E_step_atonce(all_tHMMobj, all_gammas):
+    """ perform the M_E step when all the concentrations are given at once for all the states.
+    gms is a list of arrays with size = N x K.
+    After reshaping, we will have a list of lists for each state. 
+     """
+    gms = []
+    for gm in all_gammas:
+        gms.append(np.vstack(gm))
```

Comment:
Why `vstack` one array?

Reply:
In this, I followed the pattern that was being done for each population, only here I have the data of concentrations appended in a list, so I looped over those inside the list. I don't think it is one array; it should be the matrix that is keeping the N x K gammas for each lineage and this should be stacking all the lineages of the same condition.